### PR TITLE
 doc: releases: Document board and driver changes in 2.0 release note

### PR DESCRIPTION
**Pull request by MaureenHelm**
- Documents all the boards added and removed across all architectures
since the 1.14.0 release.
- Documents major driver changes (additions, removals, fixes, etc.) across
all driver families since the 1.14.0 release.

### CODE REVIEW

File: doc/releases/release-notes-2.0.rst [hunk 1]
```diff
@@ -38,10 +38,235 @@ Boards & SoC Support
...
+
+* Added support for the following shield boards:
+
+  * frdm_cr20a
+  * link_board_can
+  * sparkfun_sara_r4
+  * wnc_m14a2a
+  * x_nucleo_iks01a3
+
+* Removed support for the following boards:
```
Comment (dbkinder) [inline doc/releases/release-notes-2.0.rst]: I presume support for these boards still exists in the LTS release (1.14), but will no longer be supported starting with this  2.0 release.  (We should mention something like this here.)

File: doc/releases/release-notes-2.0.rst [hunk 2]
```diff
@@ -38,10 +38,235 @@ Boards & SoC Support
...
+* Ethernet
+
+  * Added LiteEth driver
+
+* Flash
+
+  * Removed Quark SE C1000 driver
+  * Removed support for Quark D2000 from QMSI driver
+  * Added STM32G0XX and STM32WB support to STM32 driver
+  * Added RV32M1 and native posix drivers
```
Comment (dbkinder) [inline doc/releases/release-notes-2.0.rst]: should be spelled POSIX, right?

File: doc/releases/release-notes-2.0.rst [hunk 3]
```diff
@@ -38,10 +38,235 @@ Boards & SoC Support
...
+  * Removed support for Quark SE C1000 from DesignWare driver
+  * Added TI CC13xx / CC26xx driver
+  * Implemented ASYNC API in SAM0 driver
+
+* Timer
+
+  * Added Xilinx ZynqMP PS ttc driver
+  * Added support for SMP to ARC V2 driver
+  * Added MEC1501 32 KHZ, local APIC timer, and LiteX drivers
+  * Replaced native posix system timer driver with tickless support
```
Comment (dbkinder) [inline doc/releases/release-notes-2.0.rst]: POSIX

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: doc/releases/release-notes-2.0.rst (hunk 1 above)
Comment (MaureenHelm) [reply doc/releases/release-notes-2.0.rst]: Yes, these boards still exist in the LTS release. Do you think we need to mention this for everything we remove in the 2.0 release, or just for boards?

**Review by dbkinder — APPROVED** [review]
Thanks for the spelling fixes. We could discuss how to properly deprecate board support (it was a rather sudden removal), but for now, we can merge this PR (so it won't conflict with subsequent PRs) and work on messaging later.

**Review by ioannisg — APPROVED** [review]

**Pull request merged**